addons/binding/org.openhab.binding.fronius/src/main/java/org/openhab/binding/fronius/handler/FroniusHandler.java

```diff
@@ -0,0 +1,369 @@
+            case FroniusBindingConstants.InverterDataChannelIac:
+                return inverterRealtimeResponse.getBody().getData().getIac();
+            case FroniusBindingConstants.InverterDataChannelIdc:
+                return inverterRealtimeResponse.getBody().getData().getIdc();
+            case FroniusBindingConstants.InverterDataChannelUac:
+                return inverterRealtimeResponse.getBody().getData().getUac();
+            case FroniusBindingConstants.InverterDataChannelUdc:
+                return inverterRealtimeResponse.getBody().getData().getUdc();
+        }
+
+        if (powerFlowResponse == null) {
+            return null;
+        }
+        switch (fieldName) {
+            case FroniusBindingConstants.PowerFlowpGrid:
+                return powerFlowResponse.getBody().getData().getSite().getPgrid();
+            case FroniusBindingConstants.PowerFlowpLoad:
+                return powerFlowResponse.getBody().getData().getSite().getPload();
+            case FroniusBindingConstants.PowerFlowpAkku:
+                return powerFlowResponse.getBody().getData().getSite().getPakku();
+        }
+
+        return null;
+    }
+
+    /**
+     * Get new data
+     *
+     */
+    private void updateData() {
+        FroniusConfiguration config = getConfigAs(FroniusConfiguration.class);
+        inverterRealtimeResponse = getRealtimeData(config.ip, config.deviceId);
+        powerFlowResponse = getPowerFlowRealtime(config.ip);
+    }
+
+    /**
+     * Make the PowerFlowRealtimeDataRequest
+     *
+     * @param ip
+     * @return {PowerFlowRealtimeResponse}
+     */
+    private PowerFlowRealtimeResponse getPowerFlowRealtime(String ip) {
+        PowerFlowRealtimeResponse result = null;
+        boolean resultOk = false;
+        String errorMsg = null;
+
+        try {
+
+            String location = PowerFlowRealtimeData.replace("%IP%", StringUtils.trimToEmpty(ip));
+            logger.debug("URL = {}", location);
+            URL url = new URL(location);
+            URLConnection connection = url.openConnection();
+
+            try {
+                String response = IOUtils.toString(connection.getInputStream());
+                logger.debug("aqiResponse = {}", response);
+                result = gson.fromJson(response, PowerFlowRealtimeResponse.class);
+            } finally {
+                IOUtils.closeQuietly(connection.getInputStream());
+            }
+
+            if (result == null) {
+                errorMsg = "no data returned";
+            } else if (result.getBody() != null) {
+                resultOk = true;
+            } else {
+                errorMsg = "missing data sub-object";
+            }
+
+            if (!resultOk) {
+                logger.warn("Error in fronius response: {}", errorMsg);
+            }
+
+        } catch (MalformedURLException e) {
+            errorMsg = e.getMessage();
+            logger.warn("Constructed url is not valid: {}", errorMsg);
```

Comment:
Aren't these real error cases because the binding does not work without a valid Url, right?